### DISCUSSION TITLE
BIRT and targetplatforms

### DISCUSSION THREAD

**Discussion by ChrisLewold** [eclipse-birt/birt#790] [General]
Greetings! 

Since years we use BIRT as part of a RCP application, which is built based on a targetplatform. 

We are glad to see BIRT coming back to life again, so we changed our targetplatform to use 
BIRT version 4.9.0.v202112301813 taken from http://download.eclipse.org/birt/update-site/4.9

This worked fine, until our build suddenly broke. 
Checking the update site we saw, that above BIRT version suddenly vanished - instead there now is 4.9.0.v202201232020. 

We assume you deployed a new build to the update site, and removed the old version. 
Doing so however kills the targetplatforms of your users, and maintaining target files is tedious work anyways. We'd really like to avoid messing around with our target files every time you provide a new version. 

Long story short: could you please keep the old versions of your builds on the update sites? 

With best regards, 
  Chris Lewold

**Comment by gab1one** [comment]
I ran into the same issue and we ended up mirroring a snapshot to our infrastructure to deal with this.

**Comment by wimjongman** [comment]
BIRT 4.9 is a flying target. However, it will contain only stable builds that have passed all our tests. So your target platform should not fail. Can you tell me what issues you see?

I do not advise you to make any of the 4.9 builds "permanent" until we have released 4.9.

**Reply by wimjongman** [reply]
By the way, you should not put a version in the target platform entries unless you explicitly require one. Just put 0.0.0 as the version and the target platform will pick the latest version.

**Comment by ChrisLewold** [comment]
Hello Wim! 

Thanks for the quick response. 
First of all we didn't recognize that the update site also contains snapshot builds, we thought it's already a final 4.9 version - surely a mistake on our side. We wanted to go to 4.9 "fast", as the previous version we used caused many problems (references to totally outdated libraries etc.) - so we were more than happy to find the 4.9 update site and thought content is final. 

We use "hard versions" on purpose. Our product uses like "don't know - let's say 100" libraries/plugins etc. 
We want to be sure that our build is 100% reproducible. In case we always take the latest version of "whatever library", then the same build today and next week would return a different result, just because a newer version of a bundle/feature became available on a given update site. And in case a bug is introduced by this new plugin version, we would get it "automatically". 

For BIRT however this approach likely makes sense, as there is no final 4.9 version yet - so we will use "0.0.0" until there is one .... (sound's a bit like using maven snapshot builds). 

Best regards, 
  Chris

**Reply by ChrisLewold** [reply]
Ah btw - we didn't see any BIRT issues. The issue was, that our target platform couldn't be resolved, as the 4.9.0.v202112301813 no longer was there ...
